src/Shared/Implementation/TelemetryConfigurationOptionsSetup.cs

```diff
@@ -65,6 +65,11 @@ public void Configure(TelemetryConfiguration configuration)
                     configuration.InstrumentationKey = this.applicationInsightsServiceOptions.InstrumentationKey;
                 }
 
+                if (this.applicationInsightsServiceOptions.ConnectionString != null)
```

Comment:
we read endpoint address further down this class. That should be moved before ConString. Else it'll override connectionstring's endpoint address.